src/Workspaces/Core/MSBuild.BuildHost/BuildHost.cs

```diff
@@ -105,7 +106,11 @@ private bool TryEnsureMSBuildLoaded(string projectOrSolutionFilePath)
         }
     }
 
+#if NET472 || NET6_0 // If we're compiling against net472 or net6.0, we get our MemberNotNull from the workspaces assembly. It has it in the net6.0 case since we're consuming the netstandard2.0 version of Workspaces.
+    [workspaces::System.Diagnostics.CodeAnalysis.MemberNotNull(nameof(_buildManager))]
+#else // If we're compiling against net7.0 or higher, then we're getting it staright from the framework.
```

Comment:
I absolutely hate this. If I don't qualify this, then we end up with conflicts since this exists in both the Workspaces DLL (visible via IVT) and also the System.Runtime reference that was getting pulled in some way. I wish there was a way for me to say just-pick-one::MemberNotNull() and the compiler picks one arbitrarily. @jaredpar told me that if I mention this to @333fred and @chsienki that this might make them sad enough to want to address this.

Longer term, the plan is to remove the reference on the Workspaces DLL entirely, since it's mostly just being used for attributes like this, or for our Contract type. Removing the reference entirely would slim down our deploy size and also just generally remove tangles like this. Getting rid of the compiler layer reference too ensures that this is just a simple isolated process with absolutely no references to get messed up with something else.